### PR TITLE
Make universal wheels work

### PR DESCRIPTION
**Pull request by hawkowl**
merged forward + fixed version of #678 by @agronholm

### CODE REVIEW

**Comment by hawkowl** [comment]
I think that this should work. I've done some tests and it *appears* to function fine on older setuptools + newer ones, without *too* much duplication.

**Comment by agronholm** [comment]
Um, so you're saying wheels created with this also install on older pip versions? Because that was my only concern with my other PR.

**Comment by agronholm** [comment]
Are you sure you've seen my alternative PR? https://github.com/crossbario/crossbar/pull/677

**Comment by hawkowl** [comment]
Old pips won't download wheels from PyPI by default -- so, it'll break if you manually install a wheel, but that's not really something you should do...

**Comment by agronholm** [comment]
How old pip versions are we talking about?

**Comment by hawkowl** [comment]
```
Python 3.4.3+ (default, Oct 14 2015, 16:03:50) 
[GCC 5.2.1 20151010] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pip
>>> pip.__version__
'1.5.6'
>>> import setuptools
>>> setuptools.__version__
'18.4'
>>> import wheel
>>> wheel.__version__
'0.26.0'
```

**Comment by agronholm** [comment]
I just tried pip 1.5.4 and it downloads wheels by default. Ubuntu Trusty ships with setuptools 0.7.2 too.

**Comment by hawkowl** [comment]
I installed those two specific versions, and it appears to work fine (on cpy, it installs wsaccel, and pypy it does not).

**Comment by agronholm** [comment]
I will take a better look at this once I'm properly rested.

File: setup.py [hunk 1]
```diff
@@ -95,27 +106,56 @@
...
-    # setproctitle does not provide wheels (https://github.com/dvarrazzo/py-setproctitle/issues/47) => disable on Windows
-    install_requires.append('setproctitle>=1.1.9')  # BSD license
 
-if sys.platform == 'win32':
-    install_requires.append('pypiwin32>=219')       # PSF license
+setproctitle_version = "1.1.9"  # BSD license
+pypiwin32_version = "219"  # PSF license
+pyinotify_version = "0.9.6"  # MIT license
+wsaccel_version = "0.6.2"  # Apache 2.0
+ujson_version = "1.33"  # BSD license
```
Comment (agronholm) [inline setup.py]: Why are the versions separate from the requirement lines? They weren't so in the original nor in my PRs and I don't see why they should be.

Comment (hawkowl) [inline setup.py]: So that when I do a version bump, I don't accidentally require two different versions depending on if your setuptools version :P

Comment (agronholm) [inline setup.py]: Good point.

**Pull request merged**